src/error_cause.rs

```diff
@@ -102,6 +102,10 @@ impl ErrorCause {
         }
 
         let value_length = len as usize - ERROR_CAUSE_HEADER_LENGTH;
+        if buf.len() < value_length {
```

Comment:
I might be wrong here, but think `buf.len()` should be greater or equal to `len`. like `buf` contains a header and value. therefore buf.len() >= {ERROR_CAUSE_HEADER_LENGTH} + {len - ERROR_CAUSE_HEADER_LENGTH} => buf.len() >= len

Reply:
I think you are correct, that check was not as strict as it could be. Fixed it.